cps2cmake

```diff
@@ -256,7 +256,10 @@ def print_targets_config(package):
             hints = [fixup_path(p) for p in requirement_info.hints]
             fd_args += ' HINTS "' + '" "'.join(hints) + '"'
 
+        print_cmake_extensions(requirement_info)
         print('find_dependency(%s%s)' % (requirement_name, fd_args))
+        for item in get_extension(requirement_info, 'X-CMake-Variables', []):
+            print('unset(%s)' % (item))
```

Comment:
One caveat: this will do Bad Things if the variable is something like `CMAKE_MODULE_PATH`. Is that okay? (Is it expected that this would not be used with such variables?)

Reply:
Heh.  I actually very specifically want to use this to set `CMAKE_MODULE_PATH`, so maybe this isn't the best idea :).  What's your advice here?  Just don't do the unset?

Reply:
This all descends from this PR: https://github.com/RobotLocomotion/drake/pull/7323 .  To summarize (since it is very long): I have a library (libdrake) that depends on another library (Protobuf), and we need the cmake file, generated from the `drake.cps` file, to reflect that.    However, we want to install a newer version of `FindProtobuf.cmake` because the version that ships with Ubuntu 16.04 is buggy.  Thus, we want to use the MODULE version of `find_package`, but when you use MODULE, you cannot also use HINTS.  Thus, @jamiesnape suggested that we directly manipulate CMAKE_MODULE_PATH to accomplish this.  Other thoughts on how to do this are welcome.

Reply:
Ah... actually, for that case (sorry!), having global-but-before variables is probably better.

Does this work for you?

```diff
diff --git a/cps2cmake b/cps2cmake
index 724bdca..3c9cc75 100755
--- a/cps2cmake
+++ b/cps2cmake
@@ -236,10 +236,13 @@ def print_convenience_rules(package):
     print('')
 
 #------------------------------------------------------------------------------
-def print_cmake_extensions(package):
-    for item in get_extension(package, 'X-CMake-Includes', []):
+def print_cmake_extensions(package, ext_template = 'X-CMake-{}'):
+    incvar = ext_template.format('Includes')
+    for item in get_extension(package, incvar, []):
         print('include(%s)' % item)
-    for item in get_extension(package, 'X-CMake-Variables', {}).iteritems():
+
+    varvar = ext_template.format('Variables')
+    for item in get_extension(package, varvar, {}).iteritems():
         print('set(%s "%s")' % item)
 
 #------------------------------------------------------------------------------
@@ -247,6 +250,8 @@ def print_targets_config(package):
     print(block_header)
     print(block_get_prefix) # TODO account for install subdirectory
 
+    print_cmake_extensions(package, 'X-CMake-{}-Init')
+
     for requirement_name, requirement_info in package.requires.iteritems():
         fd_args = ''
         if requirement_info.version is not None:
```

Reply:
...although, this might make it tricky to append to whatever `CMAKE_MODULE_PATH` has before, which seems like a plausible argument for going with the per-dependency approach after all (with logic to restore the old value after). Which way do you prefer?

Reply:
Don't worry about it; I should have explained my exact use case more when I opened up the PR.  As far as the diff goes, this seems to work just fine for me, and I like the name "Init" better than "Prepend".  Thus, I would suggest going ahead with that patch, and feel free to close out this PR.  Thank you so much!